lib/operators.js

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2023 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const { allocAndEncode, seekKey2, iterate } = require('bipf')
+const { equal } = require('jitdb/operators')
+
+const B_VALUE = allocAndEncode('value')
+const B_CONTENT = allocAndEncode('content')
+const B_TANGLES = allocAndEncode('tangles')
+const B_ROOT = allocAndEncode('root')
+const B_RECPS = allocAndEncode('recps')
+
+const B_TANGLE_MAP = new Map()
+// tangle: StrinG => B_TANGLE: Buffer
+
+module.exports = {
+  tangleRoot(tangle, value) {
+    if (!B_TANGLE_MAP.has(tangle)) {
+      B_TANGLE_MAP.set(tangle, allocAndEncode(tangle))
+    }
+    const B_TANGLE = B_TANGLE_MAP.get(tangle)
+
+    function seekTangleRoot (buffer, start, p) {
+      if (p < 0) return -1
+      p = seekKey2(buffer, 0, B_VALUE, 0)
+      if (p < 0) return -1
+      p = seekKey2(buffer, p, B_CONTENT, 0)
+      if (p < 0) return -1
+      p = seekKey2(buffer, p, B_TANGLES, 0)
+      if (p < 0) return -1
+      p = seekKey2(buffer, p, B_TANGLE, 0)
+      if (p < 0) return -1
+      return seekKey2(buffer, p, B_ROOT, 0)
+    }
+
+    return equal(seekTangleRoot, value, {
+      // prefix: 32,
+      // QUESTION (mix): what is this prefix, why can't I use it
```

Comment:
target_start is similar to start, an index into the target so that you can compare only say the last part of something. This is rarely used and could probably have used a default argument to 0. Basically you have a window into the buffer starting at `start` and checking if the bytes matches those of target (or part if target start is used).

Reply:
Prefix indexes are documented here: https://github.com/ssbc/jitdb#prefix-indexes. It is a bit strange that it was not working for you. Havn't run the code to debug.

Reply:
Remove those 2 commented out lines.